src/xacro/__init__.py

```diff
@@ -431,11 +429,11 @@ def process_include(elt, macros, symbols, func):
             # restore filestack
             restore_filestack(oldstack)
 
-        except XacroException as e:
-            if e.exc.strerror == "No such file or directory" and optional is True:
-                continue
-            else:
-                raise
+    except XacroException as e:
+        if e.exc and isinstance(e.exc, IOError) and e.exc.strerror == "No such file or directory" and optional is True:
```

Comment:
My aim was to avoid the string comparison.

Reply:
Done